redis-persistence/src/main/java/com/netflix/conductor/redis/dynoqueue/ConfigurationHostSupplier.java

```diff
@@ -34,9 +37,80 @@ public ConfigurationHostSupplier(RedisProperties properties) {
         this.properties = properties;
     }
 
+    private List<Host> loadHosts() {
+        List<Host> hostList = parseHostsFromConfig();
+        List<Host> successfulHosts = new ArrayList();
+        for (Host host : hostList) {
+            if (ping(host)) {
+                successfulHosts.add(host);
+            }
+        }
+        log.debug("Successful redis hosts after ping {}", successfulHosts);
+        return checkForMajority(hostList, successfulHosts);
+    }
+
     @Override
     public List<Host> getHosts() {
-        return parseHostsFromConfig();
+        return loadHosts();
+    }
+
+    private List<Host> checkForMajority(List<Host> allHostsList, List<Host> successfulHosts) {
+        if (allHostsList.isEmpty()) {
+            return allHostsList;
+        }
+        int majority = allHostsList.size() / 2 + 1;
+        log.debug(
+                "Successful dynomite hosts size {} allHostSize {} majority {}",
+                successfulHosts.size(),
+                allHostsList.size(),
+                majority);
+        if (successfulHosts.size() >= majority) {
+            return successfulHosts;
+        } else {
+            log.info("Successful dynomite hosts {}", successfulHosts);
+            log.info("Configured dynomite hosts {}", allHostsList);
+            throw new RuntimeException(
+                    "Successful dynomite hosts size is less than majority. Hence conductor will not connect to redis");
+        }
+    }
+
+    private boolean ping(Host host) {
+        JedisPool jedisPool = null;
+        Jedis jedis = null;
+        try {
+            JedisPoolConfig config = new JedisPoolConfig();
+            config.setMinIdle(1);
+            config.setMaxTotal(1);
+            if (host.getPassword() == null) {
+                jedisPool = new JedisPool(config, host.getHostName(), host.getPort());
```

Comment:
in this case what is the timeout value? I guess in the second case the timeout is 3000 ms? Can you make it consistent

Reply:
fixed